### PR TITLE
Pull request for Issue2069: Add scan serial number to AMScanThumbnailGridView scan items.

### PR DESCRIPTION
**Pull request by mButlerDM**
Added method to generate a QRect for the serial number. Adjusted name rectangle so it was shifted to the right so that the serial could paint in front of it. Added calls to paint serial along with other elements.

Since the serial QModelIndex value was used to paint the background previously, new methods had to be added to generate a rectangle for the scan item background and paint it, apart from the functions dependent on the model index values.

### CODE REVIEW

**Comment by mButlerDM** [comment]
@iainworkman Since you helped with this would you mind taking a look when you have a chance?

Thanks

File: source/ui/dataman/AMScanThumbnailGridView.cpp [hunk 1]
```diff
@@ -4,415 +4,443 @@
 #include "ui/dataman/AMScanThumbnailGridInputManager.h"
 
 AMScanThumbnailGridView::AMScanThumbnailGridView(QWidget *parent) :
-	QAbstractItemView(parent)
```
Comment (iainworkman) [inline source/ui/dataman/AMScanThumbnailGridView.cpp]: Either your spacing is messed up now, or mine was messed up when I first wrote this code. My guess is the latter. You don't need to do anything with this code, but do remind me to check your settings.

File: source/ui/dataman/AMScanThumbnailGridView.cpp [hunk 2]
```diff
@@ -4,415 +4,443 @@
...
-		case 2:
-			return geometryManager_->thumbnailTitleRectangle(rowContentRect);
-
-		default:
-			return QRect();
-		}
-	}
+    QRect rowContentRect;
+    /*
+    if(!index.isValid()){
```
Comment (iainworkman) [inline source/ui/dataman/AMScanThumbnailGridView.cpp]: Can this commented code be removed?

Comment (mButlerDM) [inline source/ui/dataman/AMScanThumbnailGridView.cpp]: @iainworkman Yes this should be removed, it was a left over fragment from that first solution attempt. I removed it, tested, and pushed the update.

**Comment by iainworkman** [comment]
Sorry, forgot to leave the reviewed comment.
Looks good to me.
Time Estimate: 5 minutes.

**Comment by dretrex** [comment]
Looks good.
Time Estimate: 3 minutes

**Pull request merged**